src/python/pants_release/generate_github_workflows.py

```diff
@@ -867,9 +847,35 @@ def build_wheels_job(
             "steps": [
                 *initial_steps,
                 *([] if platform == Platform.LINUX_ARM64 else [install_go()]),
-                *helper.build_wheels(),
+                {
+                    "name": "Build wheels",
+                    "run": "./pants run src/python/pants_release/release.py -- build-wheels",
+                    "env": helper.platform_env(),
+                },
                 helper.upload_log_artifacts(name="wheels"),
                 *([deploy_to_s3("Deploy wheels to S3")] if for_deploy_ref else []),
+                {
+                    "name": "Build Pants PEX",
+                    "run": "./pants package src/python/pants:pants-pex",
```

Comment:
This is doing "build wheels, publish wheels, build pex, publish pex", which seems a little more fragile in terms of idempotency. For instance, if building the PEX fails for some reason, the release will have partially published artefacts (wheels only).

In addition, it means the `upload_log_artifacts` call no longer includes the logs from building the PEX.

What do you think about moving this step to happen before publishing/uploading anything, right after building wheels?  So "build wheels, build pex, publish wheels, publish pex", meaning the processes (building artefacts) that are more complicated/more likely to fail all happen before the simpler steps of uploading files.

Reply:
True although FWIW the wheel publishing to S3 is likely to be sunset.

However, your point stands. I'll edit.